src/core/utils/compatibility.js

```diff
@@ -165,25 +162,16 @@ export function getClientCompatibility({
 }
 
 export const isQuantumCompatible = ({ addon }) => {
-  // HACK: we use a hardcoded User Agent string corresponding to MacOS/Firefox
-  // 57.0 to determine whether the given `addon` is compatible with Quantum.
-  // The OS is not important as far as it is not `iOS`.
-  //
-  // See: https://github.com/mozilla/addons-frontend/issues/3868.
-  const userAgentInfo = UAParser(oneLine`Mozilla/5.0 (Macintosh; Intel Mac OS X
-    10.12; rv:57.0) Gecko/20100101 Firefox/57.0`);
-
-  const supportsDesktop57 = getClientCompatibility({
-    addon,
-    clientApp: CLIENT_APP_FIREFOX,
-    userAgentInfo,
-  }).compatible;
-
-  const supportsAndroid57 = getClientCompatibility({
-    addon,
-    clientApp: CLIENT_APP_ANDROID,
-    userAgentInfo,
-  }).compatible;
-
-  return supportsDesktop57 || supportsAndroid57;
+  // TODO: refactor code that inspects the real compatibility
+  // object and re-use that logic to accomplish this instead.
+  // https://github.com/mozilla/addons-frontend/issues/3814
+
+  if (!addon.current_version) {
+    return false;
+  }
+  return addon.current_version.files.some((file) => {
+    // These checks are fragile because future mozilla-signed extensions
+    // may not be Quantum compatible.
+    return file.is_webextension || file.is_mozilla_signed_extension;
+  });
```

Comment:
The `isWebExtension` computed prop was already available. How do you feel about computing `isMozillaSignedExtension` in the `reducer` then?a